packages/svelte/src/internal/client/dom/blocks/boundary.js

```diff
@@ -300,22 +355,26 @@ export class Boundary {
 			// If the failure happened while flushing effects, current_batch can be null
 			Batch.ensure();
 
-			this.#pending_count = 0;
+			// this ensures we modify the cascading_pending_count of the correct parent
+			// by the number we're decreasing this boundary by
+			this.update_pending_count(-this.#pending_count, true);
 
 			if (this.#failed_effect !== null) {
 				pause_effect(this.#failed_effect, () => {
 					this.#failed_effect = null;
 				});
 			}
 
-			this.pending = true;
+			// we intentionally do not try to find the nearest pending boundary. If this boundary has one, we'll render it on reset
+			// but it would be really weird to show the parent's boundary on a child reset.
+			this.pending = this.has_pending_snippet();
```

Comment:
Don't think this is quite right, since this uses the getter which will set the parent to `true` if the current one doesn't have a pending snippet. Did you mean to do this?
```suggestion
			this.#pending = this.has_pending_snippet();
```